homeassistant/components/lyric/climate.py

```diff
@@ -100,13 +100,12 @@ class LyricClimate(LyricDeviceEntity, ClimateEntity):
 
     def __init__(
         self,
-        hass: HomeAssistantType,
         coordinator: DataUpdateCoordinator,
         location: LyricLocation,
         device: LyricDevice,
     ) -> None:
         """Initialize Honeywell Lyric climate entity."""
-        self._temperature_unit = hass.config.units.temperature_unit
+        self._temperature_unit = self.hass.config.units.temperature_unit
```

Comment:
https://github.com/home-assistant/core/pull/39695#discussion_r567496188

@MartinHjelmare Swapping this to use `self.hass` causes:

```py
Traceback (most recent call last):
  File "/workspaces/core/homeassistant/helpers/entity_platform.py", line 198, in _async_setup_platform
    await asyncio.shield(task)
  File "/workspaces/core/homeassistant/components/lyric/climate.py", line 85, in async_setup_entry
    entities.append(LyricClimate(coordinator, location, device))
  File "/workspaces/core/homeassistant/components/lyric/climate.py", line 108, in __init__
    self._temperature_unit = self.hass.config.units.temperature_unit
AttributeError: 'NoneType' object has no attribute 'config'
```

Are you sure that this is correct? Or do I have to now pass this unit in the calling method instead?